README_EN.md

```diff
@@ -0,0 +1,240 @@
+
+### Usage
+
+#### Set the url in Xcode
+
+In Xcode, choose your Project's `Target`, then clcik `Info`, add a `URL Types`. For example:
+
+- Identifier: com.frdintent
+- URL Schemes: frdintent
+- Role: Editor
+- Icon:
+
+#### Take over control of app's url call
+
+````Swift
+  func application(app: UIApplication, openURL url: URL, options: [String : AnyObject]) -> Bool {
+    return FRDURLRoutes.sharedInstance.route(url: url)
+  }
+```
+
+#### Register
+
+Via code:
+
+```Swift
+  let routes = FRDURLRoutes.sharedInstance
+  routes.register(url: URL(string: "/story/:storyId")!, clazz: SecondViewController.self)
+```
+
+Via plist file:
+
+```Swift
+  let plistPath = Bundle.main.path(forResource: "FRDURLRoutesRegisters", ofType: "plist")
+  let routes = FRDURLRoutes.sharedInstance
+  routes.register(plistFile: plistPath)
+```
+
+Register a block handler.
+
+```Swift
+  let router = FRDURLRoutes.sharedInstance
+  router.register(url: URL(string: "/user/:userId")!) { (params: [String: Any]) in
+    let intent = Intent(url: params[URLRoutes.URLRoutesURL] as! URL)
+    if let topViewController = UIApplication.topViewController() {
+      FRDControllerManager.sharedInstance.startController(source: topViewController, intent: intent)
+    }
+  }
+```
+
+### Get url parameters
+
+FRDIntent/URLRoutes supports simple url parameter pattern match. In the example above, we register url with the pattern `"/story/:storyId"`, if there is an external call like `frdintent://frdintent.com/story/123`, FRDIntent/URLRoutes will save the key `storyId` and the value `123` into the block handler's parameter `params` that is a dictionary. We can get `123` from the dictionary `params` by key `storyId`.
+
+
+## Attention
+
+#### Prefix
+
+Prefix is not necessary in Swift project because of Swift's visiblity declaration. But considering that this library will be used widely in Objective-C project, it's better to add `FRD` prefix to avoid the name conflict in Objective-C.
+
+#### Type of `source` parameter
+
+In the class `FRDControllerManager`, the methods `startController(source: UIVieController, intent: FRDIntent)` and `startControllerForResult(source: UIViewController, intent: FRDIntent, requestCode: Int)` don't constrain the type of `source` parameter. The precise type of `source` is like what does `UIViewController<FRDIntentReceivable>` or `UIViewController<FRDIntentForResultReceivable>` present: it's a specific class conforming a specific protocol. This must be implemented awkwardly by generic type in Swift 3. The generic type is not precise in this situation. And it's more terrrible that the generic method in Swift cann't be used by Objective-C. So we do a tradeoff to declare the source just as UIViewController type. The user of this library must ensure that the source conform the `FRDIntentReceivable` protocol or the `FRDIntentForResultReceivable` protocol by himself.
```

Comment:
In the class `FRDControllerManager`, methods `startController(source: UIViewController, intent: FRDIntent)` and `startControllerForResult(source: UIViewController, intent: FRDIntent, requestCode: Int)` don't constrain view controller type of the `source` parameter. The precise type of `source` should be either `UIViewController<FRDIntentReceivable>` or `UIViewController<FRDIntentForResultReceivable>`, conforming with the accepted protocol.

In Swift 3, to add this constraint, we'd have to awkwardly implement some generic types, and what's even worse is that generic methods in Swift don't work in Objective-C. So the tradeoff is we only limit the `source` parameter to be a `UIViewController`, it's up to the developer to conform with the protocols and implement necessary methods as needed.